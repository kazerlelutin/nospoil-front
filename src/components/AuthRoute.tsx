import { useAuthRedirect } from '@/hooks/useAuthRedirect'
import { Route, RouteProps } from 'preact-iso'
import { JSX } from 'preact/jsx-runtime'

//SOLID
export function AuthRoute(props: RouteProps<JSX.Element>) {
  useAuthRedirect()

  return <Route {...props} />
}
