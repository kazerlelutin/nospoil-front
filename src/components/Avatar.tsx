import { forwardRef } from 'preact/compat'

type AvatarProps = {
  src: string
  alt: string
  size?: 'sm' | 'md' | 'lg'
}

export const Avatar = forwardRef(
  ({ src, alt, size = 'md' }: AvatarProps, ref) => {
    return (
      <img
        src={`${
          import.meta.env.VITE_SUPABASE_URL
        }/storage/v1/object/public/avatars/${src}`}
        //@ts-ignore
        ref={ref}
        alt={alt || 'avatar'}
        data-size={size}
        class="w-16 h-16  data-[size=sm]:w-8 data-[size=sm]:h-8 rounded-full border-solid border-3 border-white/50 "
        onError={(e) => {
          e.currentTarget.src = '/avatar.svg'
        }}
      />
    )
  }
)
