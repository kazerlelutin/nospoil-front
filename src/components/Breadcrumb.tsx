export function Breadcrumb() {
  return <div>breadcrumb</div>
}
