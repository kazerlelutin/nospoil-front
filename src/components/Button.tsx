import { JSX } from 'preact/jsx-runtime'

type ButtonProps = {
  children: JSX.Element | string
  type?: 'button' | 'submit' | 'reset'
  onClick?: () => void
  disabled?: boolean
}

//SOLID
export function Button({
  children,
  onClick,
  disabled,
  type,
  ...rest
}: ButtonProps) {
  return (
    <button
      onClick={onClick}
      type={type}
      disabled={disabled}
      data-type={type}
      class={`
        disabled:opacity-50 border-solid border-1 border-dark-text text-dark-text bg-dark-bg 
        rounded-sm px-2 py-1 cursor-pointer hover:bg-dark-text hover:text-dark-bg 
        data-[type=reset]:border-white/20 data-[type=reset]:bg-white/5
        `}
      {...rest}
    >
      {children}
    </button>
  )
}
