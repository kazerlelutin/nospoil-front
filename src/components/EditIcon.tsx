//SOLID
export function EditIcon() {
  return (
    <svg viewBox="0 0 16.933333 16.933333">
      <g>
        <g transform="translate(0.301193,-0.42314361)">
          <path d="M 13.382473,0.84628723 11.439236,2.7895245 14.38771,5.7379982 16.330947,3.7947609 Z M 10.769078,3.459682 2.5934496,11.635311 1.822805,15.354795 5.5419233,14.583785 13.717552,6.4081557 Z" />
          <path d="M 0,2.71663 V 16.933333 H 14.216703 V 7.0388467 L 13.341304,7.9142454 V 16.058451 H 0.87488198 V 3.5920287 H 9.5063963 L 10.381795,2.71663 Z m 12.452986,0 0.875399,0.8753987 h 0.01292 v 0.012919 l 0.875399,0.8759156 V 2.71663 Z" />
        </g>
      </g>
    </svg>
  )
}
