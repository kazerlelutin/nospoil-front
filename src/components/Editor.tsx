import EditorJS from '@editorjs/editorjs'
import underline from '@editorjs/underline'
import delimiter from '@editorjs/delimiter'
import Header from '@editorjs/header'
import { useId, useMemo } from 'preact/hooks'

type EditorProps = {
  id?: string
  placeholder?: string
  onChange?: (data: any) => void
  initialValue?: any
}

export function Editor({
  id: _id,
  placeholder,
  initialValue = undefined,
  onChange,
}: EditorProps) {
  const id = _id || useId()

  const editorId = useMemo(() => {
    const ed = new EditorJS({
      /**
       * Id of Element that should contain Editor instance
       */
      autofocus: true,
      placeholder,
      data: {
        blocks: JSON.parse(initialValue || '[]'),
      },
      tools: {
        // --- BLOCKS ---
        header: {
          class: Header,
          config: {
            placeholder: 'Enter a header',
            levels: [2, 3, 4],
            defaultLevel: 2,
          },
        },
        delimiter,
        // --- INLINE ---
        underline,

        // TextSpolier,
      },
      holder: id,
      onChange: () => {
        ed.save()
          .then((outputData) => {
            onChange?.(outputData.blocks)
          })
          .catch((error) => {
            console.log('Saving failed: ', error)
          })
      },
    })
    return id
  }, [])

  return <div id={editorId}></div>
}
