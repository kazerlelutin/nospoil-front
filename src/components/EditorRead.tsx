import EditorJS from '@editorjs/editorjs'
import underline from '@editorjs/underline'
import delimiter from '@editorjs/delimiter'
import Header from '@editorjs/header'
import { useId, useMemo } from 'preact/hooks'

type EditorProps = {
  blocks: string
  id?: string
}

export function EditorRead({ blocks, id: _id }: EditorProps) {
  const id = _id || useId()

  const editorId = useMemo(() => {
    new EditorJS({
      readOnly: true,
      //@ts-ignore
      data: {
        blocks: JSON.parse(blocks || '[]'),
      },

      tools: {
        // --- BLOCKS ---
        header: {
          class: Header,
          config: {
            placeholder: 'Enter a header',
            levels: [2, 3, 4],
            defaultLevel: 2,
          },
        },
        delimiter,
        // --- INLINE ---
        underline,

        // TextSpolier,
      },
      holder: id,
      minHeight: 0,
    })
    return id
  }, [blocks])

  return <div id={editorId}></div>
}
