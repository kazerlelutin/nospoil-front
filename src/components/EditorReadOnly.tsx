import { useEditorReadOnly } from '@/hooks/useEditorReadOnly'

type EditorProps = {
  blocks: string
  id?: string
}

export function EditorReadOnly({ blocks, id }: EditorProps) {
  const editorId = useEditorReadOnly({ blocks, id })
  return <div id={editorId}></div>
}
