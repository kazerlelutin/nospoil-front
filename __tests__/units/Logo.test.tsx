import { describe, it, expect, afterEach } from 'vitest'
import { cleanup, render } from '@testing-library/preact'
import { Logo } from '@/components/Logo'

describe('Logo', () => {
  afterEach(cleanup)
  it('should render', () => {
    const { container } = render(<Logo />)
    expect(container).toMatchSnapshot()
  })
})
