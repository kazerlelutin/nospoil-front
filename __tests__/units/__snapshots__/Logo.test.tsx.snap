// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > should render 1`] = `
<div>
  <svg
    class="fill-dark-text w-24"
    height="7.2583899mm"
    id="logo"
    version="1.1"
    viewBox="0 0 40.712616 7.2583899"
    width="40.712616mm"
  >
    <g
      transform="translate(-60.301705,-109.00988)"
    >
      <path
        d="m 70.495438,109.00988 c -0.999072,0 -1.851742,0.35312 -2.557983,1.05937 -0.70624,0.70624 -1.059367,1.55891 -1.059367,2.55798 0,0.99908 0.353127,1.85174 1.059367,2.55799 0.706241,0.70624 1.558911,1.05936 2.557983,1.05936 0.999073,0 1.851742,-0.35312 2.557984,-1.05936 0.70624,-0.70625 1.059367,-1.55891 1.059367,-2.55799 0,-0.99907 -0.353127,-1.85174 -1.059367,-2.55798 -0.706242,-0.70625 -1.558911,-1.05937 -2.557984,-1.05937 z m 0,2.06706 c 0.0656,0 0.129044,0.005 0.19172,0.0119 a 0.85293162,0.85293162 0 0 0 -0.202055,0.55035 0.85293162,0.85293162 0 0 0 0.853178,0.85318 0.85293162,0.85293162 0 0 0 0.648022,-0.29972 c 0.0389,0.1371 0.05943,0.28158 0.05943,0.43459 0,0.4272 -0.151586,0.79238 -0.454752,1.09554 -0.303167,0.30317 -0.66835,0.45476 -1.095541,0.45476 -0.427188,0 -0.792373,-0.15159 -1.09554,-0.45476 -0.303168,-0.30316 -0.454753,-0.66834 -0.454753,-1.09554 0,-0.42718 0.151585,-0.79237 0.454753,-1.09554 0.303167,-0.30317 0.668352,-0.45475 1.09554,-0.45475 z"
      />
      <path
        d="m 98.947276,114.17754 h 2.067044 v 2.06705 h -4.134093 v -7.23468 h 2.067049 z"
      />
      <path
        d="m 96.463719,109.00991 v 7.23468 h -2.067051 v -7.23468 z"
      />
      <path
        d="m 81.393998,109.00988 v 7.2347 h 2.067057 v -2.06705 h 1.550293 c 1.72254,0 2.583822,-0.86129 2.583822,-2.58382 0,-1.72255 -0.861282,-2.58383 -2.583822,-2.58383 z m 2.067057,2.06706 h 1.550293 c 0.344508,0 0.516764,0.17225 0.516764,0.51677 0,0.3445 -0.172256,0.51676 -0.516764,0.51676 h -1.550293 z"
      />
      <path
        d="m 77.411718,113.66078 q -2.583814,0 -2.583814,-2.34611 0,-2.30476 2.583814,-2.30476 h 3.358957 v 2.06705 h -3.358957 q -0.516763,0 -0.516763,0.23771 0,0.27906 0.516763,0.27906 h 1.550288 q 2.583813,0 2.583813,2.30476 0,2.3461 -2.583813,2.3461 h -3.358958 v -2.06705 h 3.358958 q 0.516762,0 0.516762,-0.27905 0,-0.23771 -0.516762,-0.23771 z"
      />
      <path
        d="m 62.368756,112.62725 v 3.61734 h -2.067051 v -7.23468 h 2.067051 l 2.583813,3.61734 v -3.61734 h 2.067051 v 7.23468 h -2.067051 z"
      />
      <path
        d="m 90.965763,109.03357 c -0.999072,0 -1.851742,0.35312 -2.557983,1.05937 -0.70624,0.70624 -1.059367,1.55891 -1.059367,2.55798 0,0.99908 0.353127,1.85174 1.059367,2.55799 0.706241,0.70624 1.558911,1.05936 2.557983,1.05936 0.999073,0 1.851742,-0.35312 2.557984,-1.05936 0.70624,-0.70625 1.059367,-1.55891 1.059367,-2.55799 0,-0.99907 -0.353127,-1.85174 -1.059367,-2.55798 -0.706242,-0.70625 -1.558911,-1.05937 -2.557984,-1.05937 z m 0,2.06706 c 0.0656,0 0.129044,0.005 0.19172,0.0119 a 0.85293162,0.85293162 0 0 0 -0.202055,0.55035 0.85293162,0.85293162 0 0 0 0.853178,0.85318 0.85293162,0.85293162 0 0 0 0.648022,-0.29972 c 0.0389,0.1371 0.05943,0.28158 0.05943,0.43459 0,0.4272 -0.151586,0.79238 -0.454752,1.09554 -0.303167,0.30317 -0.66835,0.45476 -1.095541,0.45476 -0.427188,0 -0.792373,-0.15159 -1.09554,-0.45476 -0.303168,-0.30316 -0.454753,-0.66834 -0.454753,-1.09554 0,-0.42718 0.151585,-0.79237 0.454753,-1.09554 0.303167,-0.30317 0.668352,-0.45475 1.09554,-0.45475 z"
      />
    </g>
  </svg>
</div>
`;
